src/objective-c/GRPCClient/private/GRPCConnectivityMonitor.m

```diff
@@ -129,6 +130,13 @@ - (nullable instancetype)initWithReachability:(nullable SCNetworkReachabilityRef
   if ((self = [super init])) {
     _reachabilityRef = CFRetain(reachability);
     _queue = dispatch_get_main_queue();
+    SCNetworkReachabilityFlags flags;
+    if (SCNetworkReachabilityGetFlags(_reachabilityRef, &flags)) {
```

Comment:
I read in some Apple documentation that `SCNetworkReachabilityGetFlags` can block, if it needs to resolve DNS. So we'd need to either ensure that this isn't called in the main thread (I suspect we can't), or just initialize to `nil`.